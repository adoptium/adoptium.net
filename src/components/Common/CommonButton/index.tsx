import React from "react";

interface CommonButtonProps {
  children: React.ReactNode;
  className?: string;
  icon?: React.ReactNode;
}

const CommonButton: React.FC<CommonButtonProps> = ({
  children,
  className = "",
  icon,
}) => {
  return (
    <button
      className={`bg-transparent mt-10 border-2 border-pink-500/0 text-white text-base  leading-6 font-normal w-[186px] h-[48px] rounded-[80px] gradient-border ${className}`}
    >
      {children}
      {icon}
    </button>
  );
};

export default CommonButton;
