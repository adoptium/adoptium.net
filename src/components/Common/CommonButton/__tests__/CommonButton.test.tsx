import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import CommonButton from "../index"

describe("CommonButton component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <CommonButton className={"test"} icon={"test"}>
        test
      </CommonButton>,
    )

    expect(container).toMatchSnapshot()
  })
})
