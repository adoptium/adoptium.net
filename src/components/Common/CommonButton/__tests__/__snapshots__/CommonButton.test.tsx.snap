// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommonButton component > should render correctly 1`] = `
<div>
  <button
    class="bg-transparent mt-10 border-2 border-pink-500/0 text-white text-base  leading-6 font-normal w-[146px] h-[48px] rounded-[80px] gradient-border test"
  >
    test
    test
  </button>
</div>
`;
